Client/Frontend/Browser/OpenWithSettingsViewController.swift

```diff
@@ -0,0 +1,134 @@
+    }()
+    private var mailSchemeEnableCache: [String:Bool] = [String:Bool]()
+    private let prefs: Prefs
+    private var currentChoice: String = "mailto"
+
+    private let BasicCheckmarkCell = "BasicCheckmarkCell"
+
+    init(prefs: Prefs) {
+        self.prefs = prefs
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        title = Strings.SettingsOpenWithSectionName
+
+        tableView.accessibilityIdentifier = "OpenWithPage.Setting.Options"
+
+        tableView.registerClass(UITableViewCell.self, forCellReuseIdentifier: BasicCheckmarkCell)
+        tableView.backgroundColor = UIConstants.TableViewHeaderBackgroundColor
+
+        let headerFooterFrame = CGRect(origin: CGPointZero, size: CGSize(width: self.view.frame.width, height: UIConstants.TableViewHeaderFooterHeight))
+        let headerView = SettingsTableSectionHeaderFooterView(frame: headerFooterFrame)
+        headerView.titleLabel.text = Strings.SettingsOpenWithPageTitle
+        headerView.showTopBorder = false
+        headerView.showBottomBorder = true
+
+        let footerView = SettingsTableSectionHeaderFooterView(frame: headerFooterFrame)
+        footerView.showTopBorder = true
+        footerView.showBottomBorder = false
+
+        tableView.tableHeaderView = headerView
+        tableView.tableFooterView = footerView
+
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(OpenWithSettingsViewController.SELappDidBecomeActiveNotification), name: UIApplicationDidBecomeActiveNotification, object: nil)
+    }
+
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+        reloadMailSchemeEnableCache()
+        if let prefMailtoScheme = self.prefs.stringForKey("MailToOption"), let previousChoice = mailSchemeEnableCache[prefMailtoScheme], let defaultScheme = mailSchemes[0]["scheme"] {
+            self.currentChoice = previousChoice ? prefMailtoScheme : defaultScheme
+        }
+        tableView.reloadData()
+    }
+
+    override func viewWillDisappear(animated: Bool) {
+        super.viewWillDisappear(animated)
+        self.prefs.setString(currentChoice, forKey: "MailToOption")
+    }
+
+    func SELappDidBecomeActiveNotification() {
+        reloadMailSchemeEnableCache()
+        tableView.reloadData()
+    }
+
+    func loadMailSchemes() -> [[String:String]] {
+        if let path = NSBundle.mainBundle().pathForResource("MailSchemes", ofType: "plist"), let dictRoot = NSArray(contentsOfFile: path) {
+            var schemes: [[String:String]] = []
+            dictRoot.forEach({ dict in
+                let dictionary = dict as! [String:String]
+                schemes.append(dictionary)
+            })
+            return schemes
+        }
+        return []
+    }
+
+    func reloadMailSchemeEnableCache() {
```

Comment:
Nit: I wonder if there is a way to combine the `loadMailSchemes` with the `enabledCache`. Seems like we're storing the stuff twice. Could this work?

```swift
typealias MailtoProviderEntry = (name: String, scheme: String, enabled: Bool)
let mailProviderSource = [MailtoProviderEntry]()

...

mailProviderSource = dictRoot.map {  name, scheme in (name, scheme, enabled: false) }
```